content/ADR-166-entities-and-crdt-modification.md

```diff
@@ -0,0 +1,98 @@
+---
+adr: 166
+title: Entity definition and deleted entities set for CRDT
+date: 2023-01-09
+status: Living
+type: Standards Track
+spdx-license: CC0-1.0
+authors:
+  - leanmendoza
+---
+
+## Abstract
+This ADR introduces the last definition of an Entity for SDK7 and also aims to change the ADR-117 CRDT specification adding a rule and a command.
+
+## Context
+
+With the precedent of the ADRs (quote the adrs), the bases for the type of data and the basic rules of consensus between peers were defined. In the framework of the ECS system, the entity is a representative number within a data collection of a certain component and this number was established as a 32-bit positive integer. Up to now, no limit has been established (neither theoretical nor practical mention) on the number of supported entities, but it is inevitable to think that 2^32 of entities is an insane number.
+
+Over the course of a scene, entities can be created and destroyed. Today, the destruction of entities involves their deprecation, since each component will have its history (timestamp > 0), even though it implies a null final state (no data in all components). It is important to highlight it so that there is no conflict in the definition of its state. For example: for the parenting of entities we use the Transform component, and suppose that in entity B we point to entity A. If we destroy entity A, the default behavior will be that entity B re-points to the root entity ( if it was not made explicit that the children of A should be destroyed). If we wanted to reuse the entity A (the number) with another "meaning", it would have an unwanted side effect, since A meant something else in the past.
```

Comment:
```suggestion
Over the course of a scene, entities can be created and destroyed. Today, the destruction of entities involves their deprecation, since each component will have its history (timestamp > 0), even though it implies a `null` final state (no data in all components). It is important to highlight it so that there is no conflict in the definition of its state. For example: for the parenting of entities we use the `Transform` component, and suppose that in entity B we point to entity A. If we destroy entity A, the default behavior will be that entity B re-points to the root entity (if it was not made explicit that the children of A should be destroyed). If we wanted to reuse the entity A (the number) with another "meaning", it would have an unwanted side effect, since A meant something else in the past.
```